tests/test_dashboard.py

```diff
@@ -0,0 +1,90 @@
+import time
+from tests.conftest import logging
+
+logger = logging.getLogger(__name__)
+
+def test_get_metadata_dashboard(api_session, base_url):
+    session = api_session
+    base_url = base_url
+    url = base_url + "/api/v2/dashboards"
+    data = {
+        "offset":0,
+        "limit":100,
+        "sort":"-timestamp"
+    }
+    response = session.get(url, params=data, headers=session.headers, verify=False)
+    time.sleep(10)
+    logger.info(f"----API Log---- {url}:::{response.status_code}::::\n{response.text}")
+    response.raise_for_status()
+
+def test_post_add_dashboard(api_session, base_url, unique_id):
+    session = api_session
+    base_url = base_url
+    unique_id = unique_id
+    url = base_url + "/api/v2/dashboards"
+    data = {
+        "name": unique_id,
+        "label": "API testing dashboard",
+        "description": "Dashboard",
+        "enabled": True,
+        "dashboard_filters": {},
+        "dashboard_sections": [
+            {
+                "title": unique_id,
+                "widgets": [
+                    {
+                        "widget_type": "label",
+                        "label": "<center><h2>API Automation</h2></center>",
+                        "min_width": 12,
+                        "max_width": 12,
+                        "height": 1
+                    }
+                ]
+            }
+        ]
+    }
+    response = session.post(url, json=data, headers=session.headers, verify=False)
+    time.sleep(10)
+    logger.info(f"----API Log---- {url}:::{response.status_code}::::\n{response.text}")
+    response.raise_for_status()
+
+def test_put_update_dashboard_data(api_session, base_url, unique_id):
+    session = api_session
+    base_url = base_url
+    unique_id = unique_id
+    url = base_url + f"/api/v2/dashboards/dashboard/{unique_id}"
```

Comment:
Having this kind of dependency (test_put_update_dashboard_data depends on execution of test_post_add_dashboard before) should be explicitly called out explicitly. Otherwise, it will fail when the testing is run using parallelization. You can research about libraries that support this.